flight/mcu/lib/teensyShared/SDLogger.cpp

```diff
@@ -0,0 +1,117 @@
+#include "SDLogger.h"
+#include "pins.h"
+#include "FS.h"
+
+SDLogger sd_logger(dataLogger); // NOLINT(cppcoreguidelines-interfaces-global-init)
+
+SDLogger::SDLogger(DataLogBuffer& buffer) : view(buffer) {}
+
+/**
+ * @brief Creates the name for a file to be written to SD card.
+ *
+ * @param fileName Pointer to char[] containing intended name of file. Do not
+ * include number or file extension at end of name. Make sure this is longer
+ * than it needs to be.
+ * @param fileExtension Pointer to char[] containing the file extension for the
+ * file.
+ * @return char* Pointer to inputted char[]. It now contains number (if
+ * duplicate file existed) and .csv file extension.
+ */
+char* sd_file_namer(char* fileName, char* fileExtensionParam) {
+    char fileExtension[strlen(fileExtensionParam) + 1];
+    strcpy(fileExtension, fileExtensionParam);
+
+    char inputName[strlen(fileName) + 1];
+    strcpy(inputName, fileName);
+
+    strcat(fileName, fileExtension);
+
+    // checks to see if file already exists and adds 1 to filename if it does.
+    if (SD.exists(fileName)) {
+        bool fileExists = false;
+        int i = 1;
+        while (!fileExists) {
+            if (i > 999) {
```

Comment:
We might be able to change this number. I think the 999 was arbitrary. Even if we keep it at the same value, create a macro/constant because magic number bad.